types/src/transaction/error.rs

```diff
@@ -229,17 +227,23 @@ impl Encodable for Error {
             Error::CannotBurnCentralizedAsset => tag_with(s, ERROR_ID_CANNOT_BURN_CENTRALIZED_ASSET),
             Error::CannotComposeCentralizedAsset => tag_with(s, ERROR_ID_CANNOT_COMPOSE_CENTRALIZED_ASSET),
             Error::InvalidDecomposedInput {
-                address,
+                asset_type,
+                shard_id,
                 got,
-            } => tag_with(s, ERROR_ID_INVALID_DECOMPOSED_INPUT).append(address).append(got),
+            } => tag_with(s, ERROR_ID_INVALID_DECOMPOSED_INPUT).append(asset_type).append(shard_id).append(got),
```

Comment:
You should change `fn list_length_for(tag: u8)`

Reply:
Fixed. There are various error types, and every implementation is different from others...